### PR TITLE
Chylli/cover

### PR DESCRIPTION
**Pull request by chylli**

### CODE REVIEW

File: Makefile.PL [hunk 1]
```diff
@@ -10,19 +10,26 @@ WriteMakefile(
                 "ExtUtils::MakeMaker" => "6.64",
             },
             BUILD_REQUIRES => {
-                'Test::More'         => 0,
-                'Test::FailWarnings' => 0,
+                'Test::More'                      => 0,
+                'Test::FailWarnings'              => 0,
+                'Devel::Cover'                    => '1.23',
```
Comment (tom-binary) [inline Makefile.PL]: These are author prereqs: they're only used for author testing, and are not a requirement for installing the module. You could put them in `TEST_REQUIRES` - but don't: use a cpanfile instead.

```
on 'develop' => sub {
 requires 'Devel::Cover', '>= 1.23';
 ...
};
```

There's a few notes on `cpanfile`+`dzil` handling in https://github.com/regentmarkets/cpan/wiki/Using-dzil - basically this:

``` bash
mymeta-cpanfile > cpanfile
```

should be enough to get an initial `cpanfile` created.

File: Makefile.PL [hunk 2]
```diff
@@ -10,19 +10,26 @@ WriteMakefile(
                 "ExtUtils::MakeMaker" => "6.64",
```
Comment (tom-binary) [inline Makefile.PL]: No need to support old `ExtUtils::MakeMaker` versions: if someone can install this module, they can upgrade existing modules as well. 6.64 dates back to 2012, so it's reasonable to list this as a requirement. That should simplify the `Makefile.PL` logic.

File: .travis.yml [hunk 3]
```diff
@@ -11,9 +11,13 @@ perl:
     - "5.14"
     - "5.12"
     - "5.10"
+env: CODECOV_TOKEN=651316b3-f2f6-40fa-babc-13e07893b94a
 before_install:
     - cpanm --installdeps .
```
Comment (tom-binary) [inline .travis.yml]: It's not too important, but `cpm` is slightly faster, see https://github.com/binary-com/perl-Date-Utility/blob/master/.travis.yml#L21 for an example.

File: .travis.yml [hunk 4]
```diff
@@ -11,7 +11,6 @@ perl:
     - "5.14"
     - "5.12"
     - "5.10"
-env: CODECOV_TOKEN=651316b3-f2f6-40fa-babc-13e07893b94a
```
Comment (tom-binary) [inline .travis.yml]: Note that we'll need to invalidate and regenerate this token, since this is a public repo.

File: Makefile.PL (hunk 1 above)
Comment (chylli) [reply Makefile.PL]: But if we use `cpanfile` and `Makefile.PL` together, will it be more confusable ?

Comment (chylli) [reply Makefile.PL]: Do you want me to change it to dzil together ?

File: .travis.yml (hunk 4 above)
Comment (chylli) [reply .travis.yml]: regenerated

**Pull request merged**